ffi/passmanagers.cpp

```diff
@@ -7,6 +15,49 @@
 
 using namespace llvm;
 
+
+/* Helpers to override the diagnostic handler when running
+ * optimizations.
+ * The default diagnostic handler prints some remarks unconditionally,
+ * see http://lists.llvm.org/pipermail/llvm-dev/2016-July/102252.html
+ */
+typedef std::tuple<LLVMContext::DiagnosticHandlerTy, void *> diag_handler_t;
+
+static diag_handler_t
+SetOptimizationDiagnosticHandler(LLVMContext &Ctx)
+{
+    auto diagnose = [] (const DiagnosticInfo &DI, void *c) {
+        // If an error, print the message and bail out (as the default
+        // handler does).
+        DiagnosticSeverity DS = DI.getSeverity();
+        if (DS == DS_Error) {
+            raw_ostream &out = errs();
+            DiagnosticPrinterRawOStream DP(out);
+            out << "LLVM error: ";
+            DI.print(DP);
+            out << "\n";
+            exit(1);
```

Comment:
`exit(1)` is what LLVM does by default. Note that `LLVMPY_RunFunctionPassManager` returns 1 if something was modified, 0 otherwise, but does not return any error status.

I'm not aware of optimizations that might produce errors, do we have examples of how to trigger such an error?